src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(student: Student): number {
+  const sum: number
+    = student.grades.reduce((accumulator: number, currentValue: number) => {
+      return accumulator + currentValue;
+    }, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const arrayForSorting = [...students];
+  const orderForSorting: number = order === 'asc' ? 1 : -1;
+
+  return arrayForSorting.sort((studentA: Student, studentB: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return studentA.name.localeCompare(studentB.name)
+          * orderForSorting;
+
+      case SortType.Surname:
+        return studentA.surname.localeCompare(studentB.surname)
+          * orderForSorting;
```

Comment:
You can simplify your code like this 
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
```

Reply:
done